scrubber/scrubber.ts

```diff
@@ -19,24 +16,55 @@ async function getConfigFile(filename: string): Promise<ScrubberConfig> {
 class Scrubber {
   tags: TagNameToAction = {};
 
-  dirs: string[] = [];
+  config?: ScrubberConfig;
 
-  async parseConfig(filename: string): Promise<void> {
+  async parseConfig(filename: string) {
     // TODO validate config (e.g.properly formed tag names)
-    const config = await getConfigFile(filename);
-    this.tags = scrubberActionsToDict(config.actions);
-    this.dirs = config.dirs;
+    this.config = await getConfigFile(filename);
+    this.tags = getAllTags(this.config.cliOptionsToActions);
   }
 
   // Scrub files
-  async start(
-    actions: ScrubberAction[],
-    isDryRun: boolean = false,
-  ): Promise<void[][]> {
-    const tags = { ...this.tags, ...scrubberActionsToDict(actions) };
+  async start(options: Options, isDryRun: boolean = false): Promise<void[][]> {
+    if (!this.config) throw new Error("Missing config.");
+
+    const tags = { ...this.tags };
+    const filesToDelete = new Set<string>();
+
+    Object.values(options).forEach((val) => {
+      if (!this.config) return;
+
+      let option: CLIOption;
+
+      if (typeof val === "boolean") {
+        if (!val) return;
+        option = "auth";
+      } else {
```

Comment:
I think it'd make more sense to handle this in the CLI code, so if auth is true then map to "auth", else null. We can make it a general rule to ignore null values in this part of the scrubber. Otherwise this is not extensible if we want to add more boolean options in the future.

Return type of `cli()` now becomes:
```
export type Options = {
  backend: BackendType;
  api: APIType;
  database: DatabaseType;
  auth: "auth" | null;
};
```